backend/src/apiserver/server/util.go

```diff
@@ -98,18 +107,20 @@ func DecompressPipelineZip(compressedFile []byte) ([]byte, error) {
 	if len(reader.File) < 1 {
 		return nil, util.NewInvalidInputErrorWithDetails(err, "Error extracting pipeline from the zip file. Empty zip file.")
 	}
-	if !isYamlFile(reader.File[0].Name) {
-		return nil, util.NewInvalidInputError("Error extracting pipeline from the zip file. Expecting a YAML file inside the zip. Got: %v", reader.File[0].Name)
-	}
-	rc, err := reader.File[0].Open()
-	if err != nil {
-		return nil, util.NewInvalidInputErrorWithDetails(err, "Error extracting pipeline from the zip file. Failed to read the content.")
-	}
-	decompressedFile, err := ioutil.ReadAll(rc)
-	if err != nil {
-		return nil, util.NewInvalidInputErrorWithDetails(err, "Error reading pipeline YAML from the zip file.")
+	for _, file := range reader.File {
+		if isPipelineYamlFile(file.Name) {
+			rc, err := file.Open()
+			if err != nil {
+				return nil, util.NewInvalidInputErrorWithDetails(err, "Error extracting pipeline from the zip file. Failed to read the content.")
```

Comment:
This error doesn't look right to me. It's the result of failed Open call. Reword?

Reply:
You're right.
I wasn't changing this code, BTW - I just indented it.

Reply:
This `.Open` call is a call to open the zipped blob reader. So the message seems to be appropriate.